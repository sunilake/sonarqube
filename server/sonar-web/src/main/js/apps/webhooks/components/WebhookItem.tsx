/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import WebhookActions from './WebhookActions';
import AlertErrorIcon from '../../../components/icons-components/AlertErrorIcon';
import AlertSuccessIcon from '../../../components/icons-components/AlertSuccessIcon';
import DateTimeFormatter from '../../../components/intl/DateTimeFormatter';
import { Webhook, WebhookDelivery } from '../../../app/types';
import { translate } from '../../../helpers/l10n';

interface Props {
  onDelete: (webhook: string) => Promise<void>;
  onUpdate: (data: { webhook: string; name: string; url: string }) => Promise<void>;
  webhook: Webhook;
}

export default function WebhookItem({ onDelete, onUpdate, webhook }: Props) {
  return (
    <tr>
      <td>{webhook.name}</td>
      <td>{webhook.url}</td>
      <td>
        <LatestDelivery latestDelivery={webhook.latestDelivery} />
      </td>
      <td className="thin nowrap text-right">
        <WebhookActions onDelete={onDelete} onUpdate={onUpdate} webhook={webhook} />
      </td>
    </tr>
  );
}

export function LatestDelivery({ latestDelivery }: { latestDelivery?: WebhookDelivery }) {
  if (!latestDelivery) {
    return <span>{translate('webhooks.last_execution.none')}</span>;
  }
  return (
    <>
      {latestDelivery.success ? <AlertSuccessIcon /> : <AlertErrorIcon />}
      <span className="spacer-left">
        <DateTimeFormatter date={latestDelivery.at} />
      </span>
    </>
  );
}
