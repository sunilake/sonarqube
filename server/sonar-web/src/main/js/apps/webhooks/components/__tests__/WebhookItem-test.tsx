/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { shallow } from 'enzyme';
import WebhookItem, { LatestDelivery } from '../WebhookItem';

const latestDelivery = {
  at: '12.02.2018',
  durationMs: 20,
  httpStatus: 200,
  id: '2',
  success: true
};

const webhook = {
  key: '1',
  name: 'my webhook',
  url: 'http://webhook.target',
  latestDelivery
};

it('should render correctly', () => {
  expect(
    shallow(
      <WebhookItem
        onDelete={jest.fn(() => Promise.resolve())}
        onUpdate={jest.fn(() => Promise.resolve())}
        webhook={webhook}
      />
    )
  ).toMatchSnapshot();
});

it('should render correctly the latest delivery', () => {
  expect(shallow(<LatestDelivery latestDelivery={undefined} />)).toMatchSnapshot();
  expect(shallow(<LatestDelivery latestDelivery={latestDelivery} />)).toMatchSnapshot();
  expect(
    shallow(
      <LatestDelivery latestDelivery={{ ...latestDelivery, httpStatus: 500, success: false }} />
    )
  ).toMatchSnapshot();
});
