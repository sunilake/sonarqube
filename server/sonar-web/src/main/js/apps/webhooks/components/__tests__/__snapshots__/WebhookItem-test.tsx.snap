// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<tr>
  <td>
    my webhook
  </td>
  <td>
    http://webhook.target
  </td>
</tr>
`;
